core/node/proof_data_handler/src/lib.rs

```diff
@@ -46,18 +82,35 @@ pub async fn run_server(
                         .await
                 },
             ),
-        );
-
-    axum::Server::bind(&bind_address)
-        .serve(app.into_make_service())
-        .with_graceful_shutdown(async move {
-            if stop_receiver.changed().await.is_err() {
-                tracing::warn!("Stop signal sender for proof data handler server was dropped without sending a signal");
-            }
-            tracing::info!("Stop signal received, proof data handler server is shutting down");
-        })
-        .await
-        .context("Proof data handler server failed")?;
-    tracing::info!("Proof data handler server shut down");
-    Ok(())
+        )
+        .route(
+            "/tee_proof_inputs",
+            post(
+                move |payload: Json<TeeProofGenerationDataRequest>| async move {
+                    get_tee_proof_gen_processor
+                        .get_proof_generation_data(payload)
+                        .await
+                },
+            ),
+        )
+        .route(
+            "/submit_tee_proof/:l1_batch_number",
+            post(
```

Comment:
Given that TEE is optional, I think it probably makes sense to:

a) Rework the routing scheme so that instead of `/tee_proof_inputs`, `/submit_tee_proof`, and `/register_tee_attestation` we have `/tee/proof_inputs`, `/tee/submit_proofs` and `/tee/register_attestation` (e.g. we introduce a `/tee` namespace instead of maintaining a flat structure). This also feels more RESTful.
b) Make it configurable (e.g. it should be possible to run proof data handler _without_ tee endpoints, and tee endpoints should be opt-in).

Reply:
(If we go this way the next logical thing would be to introduce something like this for the ZK proofs, e.g. `/zk/proof_generation_data` etc), but it's certainly a bigger task and out of scope here.
cc @RomanBrodetski @EmilLuta

Reply:
@popzxc Do you think TEE endpoints/code should be optional at the compilation (`#[cfg(feature = "tee")]`), or at the runtime level? (I presume runtime)

Reply:
Runtime is fine.

Reply:
[Fixed](https://github.com/matter-labs/zksync-era/commit/dd719e16f0bfe6f605ef939c9566642514df3f57)